### PR TITLE
Disable only previously enabled addons

### PR DESCRIPTION
**Pull request by jakolehm**
It was annoying to see (and wait) "disabling addon ..." messages on every run. This PR adds a bit more logic to `AddonManager` so that we only disable addon if it was previously enabled.

PR also includes a bit more clean cluster context implementation.

### CODE REVIEW

**Comment by jakolehm** [comment]
@SpComb @kke PTAL

**Comment by SpComb** [comment]
The new `pharos-config` should only be written after the addons have been cleaned up... otherwise any failure between the `Store cluster YAML` and the `Disabling addon ...` will cause those addons to be left running.

File: lib/pharos/phases/configure_client.rb [hunk 1]
```diff
@@ -5,19 +5,55 @@ module Phases
...
+      end
+
       def config_dir
         File.join(Dir.home, '.pharos')
       end
+
+      # @param configmap [Kubeclient::Resource]
+      # @return [Pharos::Config]
+      def build_config(configmap)
+        cluster_config = YAML.safe_load(configmap.data['cluster.yml'])
```
Comment (kke) [inline lib/pharos/phases/configure_client.rb]: `Pharos::YamlFile.new(StringIO.new(configmap.data['cluster.yml']), override_filename: "#{@host}:cluster.yml").load`

**Comment by jakolehm** [comment]
> The new pharos-config should only be written after the addons have been cleaned up... otherwise any failure between the Store cluster YAML and the Disabling addon ... will cause those addons to be left running.

I agree, now config is saved after addons have completed.

```
==> Label nodes @ 46.101.146.73 46.101.96.46
    [46.101.146.73] No labels set ... 
    [46.101.96.46] No labels set ... 
==> Configuring addons ...
==> Enabling addon ingress-nginx
==> Store cluster YAML @ 46.101.146.73
    [46.101.146.73] Storing cluster configuration to configmap
==> Cluster has been crafted! (took 1 minute)
```

File: lib/pharos/phases/configure_client.rb (hunk 1 above)
Comment (jakolehm) [reply lib/pharos/phases/configure_client.rb]: why? 😃

Comment (kke) [reply lib/pharos/phases/configure_client.rb]: not any prettier but will use the same yaml-loading rules as everything else (altough the config is from `YAML.dump(JSON.parse(JSON.dump(config_hash)))` and shouldn't therefore include any trickery)

Comment (jakolehm) [reply lib/pharos/phases/configure_client.rb]: Fixed.

**Pull request merged**